src/main/webapp/WEB-INF/view/profile.jsp

```diff
@@ -119,15 +119,11 @@ User owner = (User) request.getAttribute("owner");
             <div id="chat">
                 <ul>
                     <%
-                        for (Message message : messages) {
-                            String creationTime =
-                                DateTimeFormatter.ofPattern("EEE MMM dd HH:mm:ss zzz yyyy")
-                                    .withZone(ZoneId.systemDefault())
-                                    .format(message.getCreationTime());
+                        for (String displayTime : messageDisplayTimeToMessageContent.keySet()) {
                     %>
                         <li>
-                            <strong><%= creationTime %>:</strong>
-                            <%= message.getContent() %></li>
+                            <strong><%= displayTime %>:</strong>
+                            <%= messageDisplayTimeToMessageContent.get(displayTime) %></li>
```

Comment:
I'm unfamiliar with ImmutableMaps, but do they somehow handle if two messages have the same creation time?

Reply:
In ProfileServlet.getMessageDisplayTimeToMessageContentOfUser(), I put merger function that handles the case when a duplicate key gets put. I, however, simplify the construction process, but still have a merger function in the stream processing.